### PR TITLE
CORE-7269 Storage of MGM group parameters in UTXO ledger

### PR DESCRIPTION
**Pull request by vlajos**
CORE-7269 Support persisting MGM group parameters in vault.

Runtime: https://github.com/corda/corda-runtime-os/pull/3540

### CODE REVIEW

File: data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml [hunk 1]
```diff
@@ -190,6 +190,29 @@
         <addPrimaryKey columnNames="transaction_id, file_checksum"
                        constraintName="utxo_transaction_cpk_pkey"
                        tableName="utxo_transaction_cpk"/>
+
+        <comment>
+            Keep utxo_group_parameters's relevant fields in sync with vnode-vault-creation-v1.0.xml's vnode_group_parameters.
+        </comment>
+
+        <createTable tableName="utxo_group_parameters">
```
Comment (lankydan) [inline data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:198]: Do we need a timestamp/created column like we have in most of our other tables?

Comment (vlajos) [reply data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:198]: For consistency we could add. But the date itself can reflect only when we saved the record.
If there  are any dates inside the parameters, we won't be consistent with them.
Also the records coming from back chain resolution will have different dates in the different databases.  (Or we need to spend some time to stick the first date to the record when we send it forth and back.)

`vnode_group_parameters` which is roughly the same have an epoch field.

File: data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/FindSignedGroupParametersResponse.avsc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "doc": "Response format for FindSignedGroupParametersResponse",
+  "namespace": "net.corda.data.ledger.persistence",
+  "fields": [
+    {
+      "name": "results",
+      "doc": "0 or 1 SignedGroupParameters depending on the success of the query.",
+      "type": {
+        "type": "array",
+        "items": "net.corda.data.membership.SignedGroupParameters"
+      }
```
Comment (blsemo) [inline data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/FindSignedGroupParametersResponse.avsc:13]: Does AVRO not do a nullable type, so could just return `SignedGroupParameters` or `null`?

File: data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml (hunk 1 above)
Comment (blsemo) [reply data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:198]: I think a timestamp column would be good, and if it's just for as-of queries. On the other hand maybe not essential as we don't expect this to change a lot, and I can't foresee paged queries of group paramters.

**Review by blsemo — DISMISSED** [review]
It's ok, but I have a couple of question/comments.

File: data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/FindSignedGroupParametersResponse.avsc (hunk 2 above)
Comment (vlajos) [reply data/avro-schema/src/main/resources/avro/net/corda/data/ledger/persistence/FindSignedGroupParametersResponse.avsc:13]: I did not check that. Our EntityResponse worked the same way.
Also we'll may need a way to query multiple ones in the same go.
So we can just reuse this container format.

File: data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml (hunk 1 above)
Comment (vlajos) [reply data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v1.0.xml:198]: OK. I have just added it.

**Review by blsemo — APPROVED** [review]
LGTM

**Pull request merged**